components/GameForm.jsx

```diff
@@ -1,20 +1,31 @@
 import { useRouter } from "next/router";
 import styled from "styled-components";
 
-export default function GameForm({ onSubmit }) {
+export default function GameForm({ onSubmit, selectedGame, formMode }) {
   const router = useRouter();
 
   function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
-    const data = Object.fromEntries(formData);
+    let data = Object.fromEntries(formData);
+
+    if (formMode === "edit") {
+      data = { id: selectedGame.id, progress: selectedGame.progress, ...data };
+    }
+
     onSubmit(data);
   }
 
   return (
     <StyledForm onSubmit={handleSubmit}>
       <label htmlFor="gameTitleInput">Title:</label>
-      <input type="text" id="gameTitleInput" name="title" required />
+      <input
+        type="text"
+        id="gameTitleInput"
+        name="title"
+        required
+        defaultValue={formMode === "edit" ? selectedGame.title : ""}
```

Comment:
```suggestion
        defaultValue={selectedGame?.title || ""}
```
this would be a shortcut with the same effect you could use if you like.

Reply:
we see your point but for us it seem more readable.